apps/mobile/src/libs/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import { RadioGroup as RNRadioGroup } from "react-native-radio-buttons-group";
+
+import { Text } from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import { globalStyles } from "~/libs/styles/styles";
+import { type RadioButtonProps, type ViewStyle } from "~/libs/types/types";
+
+import { DefaultStyles } from "./libs/constants/constants";
+import { styles } from "./styles";
+
+type RadioGroupItemStyle = {
+	circleBorderColorActive?: string;
+	circleBorderSize?: number;
+	circleBorderSizeActive?: number;
+	circleColorActive?: string;
+	circleSize?: number;
+} & ViewStyle;
+
+type RadioGroupItem = {
+	label: string;
+	value: string;
+};
+
+type Properties = {
+	itemContainerStyle?: RadioGroupItemStyle;
+	onChange?: (value: RadioGroupItem) => void;
+	options: RadioGroupItem[];
+};
+
+const RadioGroup: React.FC<Properties> = ({
+	itemContainerStyle = {},
+	onChange,
+	options,
+}: Properties) => {
+	const {
+		circleBorderColorActive = BaseColor.BG_BLUE,
+		circleBorderSize = DefaultStyles.DEFAULT_BORDER_SIZE,
+		circleBorderSizeActive = DefaultStyles.DEFAULT_ACTIVE_BORDER_SIZE,
+		circleColorActive = BaseColor.BG_WHITE,
+		circleSize = DefaultStyles.DEFAULT_CIRCLE_SIZE,
+	} = itemContainerStyle;
+
+	const [selectedId, setSelectedId] = useState<string>("");
```

Comment:
@fshabanov, as I understand it, this component is considered part of the form, not the form itself. Using `useAppForm` (or `useForm` from `react-hook-form`) could increase its complexity. Many features from `useAppForm` won't be used.

Do you insist I use `useAppForm` (or `useForm` from `react-hook-form`) here?

Reply:
useAppForm we use in the parent component. RadioGroup should get name, control,... via props and use useFormController (see our Input component).

Reply:
useAppForm as Arguments expect 
```
type Arguments<T extends FieldValues = FieldValues> = {
	defaultValues: DefaultValues<T>;
	mode?: keyof ValidationMode;
	validationSchema: ValidationSchema;
};
```
For Sign-in and Sign-up, we have shared validation schema. 

What do you think I should use for Onboarding validation? 

Should I install' Zod' on the mobile part or create validation in the shared folder?

Reply:
No, all the validation should be in shared. For now I believe you don't have to create form itself. Think of RadioGroup as something similar to existing Input component - it should receive control, name, etc.

Reply:
Currently, the component is done. Unfortunately, there are some uncertainties:
 - I can't commit only RadioGroup without using it because linter forces me to use component in the project; 
- I can't use RadioGroup as a form on the onboarding screen because of your previous comment. 

This is the kind of concern that I describe in the comment above.
 
One possible solution is to go back to useState instead of useAppForm and make component used in the project and create issue with this technical debt.

What do you think about it?

Reply:
Can you use [quiz-entry](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/pull/185#issue-2485208455) to temporarily apply RadioGroup?

And we still need to pass control, name, ... through props to manage the change of the field value through the component.

In the component you will be able to do:
```
    const { field } = useFormController({ name, control });
    const { value, onChange } = field;
```
And then use it something like this (maybe a little differently):
```
<RNRadioGroup
    onPress={onChange}
    selectedId={value}
...
/>
```

Reply:
This is exactly the code that I have inside RadioGroup component now.

When I use useControll inside a component, I should use useAppForm in the parent component, which should have a validationSchema.

An Example is the Input component and its usage in Sign-in or Sign-up screens. 

Do you suggest don't create separate custom RadioGroup component just write code inside [quez-entry](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/pull/185#issue-2485208455)?

Reply:
As we're implementing only the main logic of Radio Group without context we can avoid adding form-related props for now. It should be done when developing the entire screen logic, with actual form values and validation schemas. So for now we can leave the implementation with useState and then the separate issue to adopt it to usage within the form will be added.